src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -41,24 +41,32 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
         {
             // Verify the credentials are correct.
             // Have to actually ping a storage operation.
-            IStorageBlobClient client = account.CreateBlobClient();
+            IStorageQueueClient client = account.CreateQueueClient();
 
             try
             {
                 // This can hang for a long time if the account name is wrong. 
                 // If will fail fast if the password is incorrect.
-                await client.GetServicePropertiesAsync(cancellationToken);
+                await client.SdkObject.GetServicePropertiesAsync(cancellationToken);
             }
             catch (OperationCanceledException)
             {
                 throw;
             }
-            catch
+            catch (StorageException exception)
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                                        "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                if (exception.IsRemoteNameNotResolved())
```

Comment:
Rather than try to figure out via IsRemoteNameNotResolved what type of error this is, how about just changing the error message to something that handles all cases, like: "Invalid storage account '{0}'. Please make sure your credentials are correct, and that the storage account is a general purpose account and not a restricted blob storage account"? You get the idea - we just need to get the right verbiage for referencing these account types from official Azure Storage sources.